### PR TITLE
refactor: use clap for multicall of all-in-one binary

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Use `clap` builder for multi-call dispatching of all-in-one binary.

This PR also refactors the playground's profile arguments with `clap`.

Pros: perfectly consistent CLI style with other components

- All-in-one executable help
```
❯ target/debug/risingwave help      
All-in-one executable for components of RisingWave

Usage: risingwave <COMPONENT>

Components:
  compute     The worker node that executes query plans and handles data ingestion and output [aliases: compute-node, compute_node]
  meta        The central metadata management service [aliases: meta-node, meta_node]
  frontend    The stateless proxy that parses SQL queries and performs planning and optimizations of query jobs [aliases: frontend-node, frontend_node]
  compactor   The stateless worker node that compacts data for the storage engine [aliases: compactor-node, compactor_node]
  ctl         The DevOps tool that provides internal access to the RisingWave cluster [aliases: risectl]
  playground  The quick way to start a RisingWave cluster for playing around [aliases: play]
  help        Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version
```

- Component help (`risingwave frontend --help` also works)
```
❯ target/debug/risingwave help frontend
The stateless proxy that parses SQL queries and performs planning and optimizations of query jobs

Usage: risingwave frontend [OPTIONS]

Options:
      --listen-addr <LISTEN_ADDR>
          The address that this service listens to. Usually the localhost + desired port
          
          [env: RW_LISTEN_ADDR=]
          [default: 127.0.0.1:4566]
```

- Multi-call works well
```
❯ ls -l .risingwave/bin/risingwave/frontend-node
lrwxr-xr-x  1 bugenzhao  staff  74 May 23 13:40 .risingwave/bin/risingwave/frontend-node -> /Users/bugenzhao/Developer/S/risingwave-opensource/target/debug/risingwave

❯ .risingwave/bin/risingwave/frontend-node --help         
The stateless proxy that parses SQL queries and performs planning and optimizations of query jobs

Usage: frontend [OPTIONS]

Options:
      --listen-addr <LISTEN_ADDR>
          The address that this service listens to. Usually the localhost + desired port
          
          [env: RW_LISTEN_ADDR=]
          [default: 127.0.0.1:4566]
```

Cons: `RW_NODE` support is removed due to not actually being used

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
Looks nice to me.

Minor: can we print arg info?
```
❯ ./target/debug/risingwave help frontend  
Usage: risingwave frontend [args]...

Arguments:
  [args]...
```

Concern: this is a user-facing change and breaking change 🥵

**Comment by xxchan** [comment]
> So deprecation process might be needed.

Maybe not that necessary, as the error will be immediate and fixable.

**Comment by xxchan** [comment]
requested review for compatibility concern

File: src/cmd_all/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ ignored = ["workspace-hack", "workspace-config", "task_stats_alloc"]
 
 [dependencies]
 anyhow = "1"
-clap = { version = "4", features = ["derive"] }
+clap = { version = "4", features = ["cargo", "derive"] }
```
Comment (xxchan) [inline src/cmd_all/Cargo.toml:23]: Where is the `cargo` features used?

**Comment by BugenZhao** [comment]
> Minor: can we print arg info?

We can get it printed with `risingwave frontend --help`. But yes, you remind me that we can directly pass the `Opts` struct for all-in-one `Command`, instead of raw args. 😄

> Although I think (and hope) so, but we are not sure whether it's used by any user.

At least we're sure that no first-party modules from us is depending on this feature. 🥵

File: src/cmd_all/Cargo.toml (hunk 1 above)
Comment (BugenZhao) [reply src/cmd_all/Cargo.toml:23]: It's used by `command!` to read `version`, `author`, `about` etc from `Cargo.toml`.

File: src/compute/src/lib.rs [hunk 2]
```diff
@@ -39,6 +39,10 @@ use serde::{Deserialize, Serialize};
 
 /// Command-line arguments for compute-node.
 #[derive(Parser, Clone, Debug)]
+#[command(
+    version,
+    about = "The worker node that executes query plans and handles data ingestion and output"
```
Comment (BugenZhao) [inline src/compute/src/lib.rs:44]: These `about`s are summarized from our official documentation.

**Review by xxchan — APPROVED** [review]
Generally very Niiiiice to me~

**Review by xxchan — APPROVED** [review]
Generally very Niiiiice to me~

**Pull request merged**